### PR TITLE
Add timezone picker to rescheduler

### PR DESCRIPTION
**Pull request by unDemian**
**Built on top of https://github.com/Automattic/wp-calypso/pull/21732**

## Summary
This PR adds a timezone picker to the reschedule flow, which will default to the timezone selected by the customer while booking the appointment.

**More context 544-hg-gh**

The new layout looks like this: https://cloudup.com/cpwAJxgHiNF cc: @evilluendas for feedback

## Testing
**Apply code-D9600 patch and sandbox the api**
1. Book an appointment by following the steps in https://github.com/Automattic/wp-calypso/pull/21364
2. Click the reschedule link from the invitation email
3.If the email did not came through access the MC happiness/concierge page, under the `Lighthouse Only` tab you should see your appointment
4. Grab the id of your appointment
5. http://calypso.localhost:3000/me/concierge/`business-site-slug`/`id-from-email`/reschedule
6. Change the timezone
7. The time slots should reflect the new timezone
8. Reschedule the appointment
9. MC should reflect the new time slot and timezone.

### CODE REVIEW

File: client/state/data-layer/wpcom/concierge/schedules/appointments/reschedule/test/index.js [hunk 1]
```diff
@@ -21,6 +22,7 @@ describe( 'wpcom-api', () => {
 				scheduleId: 123,
 				appointmentId: 1,
 				beginTimestamp: 1234567890,
+				appointmentDetails: { meta: { timezoe: 'UTC' } },
```
Comment (southp) [inline client/state/data-layer/wpcom/concierge/schedules/appointments/reschedule/test/index.js]: typo: timezoe => timezone

File: client/me/concierge/reschedule/calendar-step.js [hunk 2]
```diff
@@ -72,17 +90,41 @@ class CalendarStep extends Component {
...
+								<Timezone
+									includeManualOffsets={ false }
+									name="timezone"
+									onSelect={ this.setTimezone }
+									selectedZone={ appointmentDetails.meta.timezone }
+								/>
+								<FormSettingExplanation>
+									{ translate( 'Choose a city in your timezone.' ) }
+								</FormSettingExplanation>
+							</FormFieldset>
```
Comment (southp) [inline client/me/concierge/reschedule/calendar-step.js]: The lines 102 ~ 103 look almost identical to [the lines 68-79 in book/info-step.js](https://github.com/Automattic/wp-calypso/blob/master/client/me/concierge/book/info-step.js#L68-L79). Would it be possible to extract it into an individual component?

**Comment by southp** [comment]
Btw, I think a rebase against the latest master should fix the wp-desktop test :)

**Comment by benhuberman** [comment]
@southp Please correct me if I missed anything -- since I didn't manage to live-test, I relied on the snippets in the code and the screenshot -- but  as far as I can see these are the relevant snippets of copy:

> Please select the desired timezone and day to reschedule your Concierge session.

Suggested tweak, flipping the order a little + and adding a tiny note of extra warmth:

> To reschedule your Concierge session, let us know your timezone and preferred day.

> What's your timezone?
Looks good to me!

> Choose a city in your timezone.
Looks good to me as well!

If there's anything else, let me know and I'd be happy to take a look!

**Comment by southp** [comment]
@benhuberman Wonderful! You have caught everything. Thanks 🙇🏻

File: client/state/data-layer/wpcom/concierge/schedules/appointments/reschedule/test/index.js (hunk 1 above)
Comment (unDemian) [reply client/state/data-layer/wpcom/concierge/schedules/appointments/reschedule/test/index.js]: Good catch, updated 👍

File: client/me/concierge/reschedule/calendar-step.js (hunk 2 above)
Comment (unDemian) [reply client/me/concierge/reschedule/calendar-step.js]: Not quite sure if we need a new component just for a form field.

**Comment by unDemian** [comment]
> To make things clearer and safer in the future, we might want to consider to move some data out of meta. I'm not 100% sure how at the moment, but I think the baseline is that updating one field should not risk wiping out other conceptually unrelated things.

Totally agree, thanks for taking the time.

File: client/me/concierge/reschedule/calendar-step.js (hunk 2 above)
Comment (mattwondra) [reply client/me/concierge/reschedule/calendar-step.js]: Good call @unDemian — I don't think this is complicated enough to merit abstraction.

Comment (southp) [reply client/me/concierge/reschedule/calendar-step.js]: Cool. I'm also good with keeping it this way. :)

**Comment by southp** [comment]
I think this is ready to land once @evilluendas having taken the final 👨‍🎨 look :)

**Pull request merged**